dozer/cogs/firstqa.py

```diff
@@ -100,6 +101,49 @@ async def frcqa(self, ctx: DozerContext, question: int):
     """
 
 
+    @commands.hybrid_command(name = "frcrule", pass_context = True)
+    @bot_has_permissions(embed_links = True)
+    @app_commands.describe(rule = "The rule number")
+    async def frcrule(self, ctx: DozerContext, rule: str):
+        letter_part = ''.join([char for char in rule if char.isalpha()])
+        number_part = ''.join([char for char in rule if char.isdigit()])
+        embed = discord.Embed(
+            title=f"Rule {letter_part.upper()}{number_part}",
+            url=f"https://frc-qa.firstinspires.org/manual/rule/{letter_part.upper()}/{number_part}",
+            color=discord.Color.blue()
+        )
+
+
+        if not re.match(r'^[a-zA-Z]\d{3}$', rule):
+
+            embed.add_field(
+                name="Error",
+                value="Invalid rule number"
+            )
+            
+        else:  
+            async with ctx.cog.ses.get('https://firstfrc.blob.core.windows.net/frc2023/Manual/HTML/2023FRCGameManual.htm') as response:
```

Comment:
I don't love hard coding this to a specific year's game manual. When this becomes 2024, it requires manual input to switch this URL. The URL before, while it has the manual embedded inside of an iframe, is year-agnostic. I have no problem with the bs4 scraping, especially for something like this, but we should keep this year agnostic.